model/src/cfd.rs

```diff
@@ -296,6 +298,35 @@ impl Order {
             self.opening_fee,
         )
     }
+
+    const OUTDATED_AFTER_MINS: i64 = 10;
+
+    /// Defines when we consider the order to be sane within the system boundaries
+    ///
+    /// This is a safety net for the taker to prevent the taker from taking an outdated order.
+    pub fn is_safe_to_take(&self, now: OffsetDateTime) -> bool {
+        !self.is_creation_timestamp_outdated(now) && self.is_oracle_event_id_sane_to_take(now)
+    }
+
+    /// Check if the the maker's offer creation timestamp is outdated
+    ///
+    /// If the creation timestamp is older than 10 minutes the offer is considered outdated
+    fn is_creation_timestamp_outdated(&self, now: OffsetDateTime) -> bool {
+        self.creation_timestamp_maker.seconds() + (Self::OUTDATED_AFTER_MINS * 60)
+            < now.unix_timestamp()
+    }
+
+    /// Check the oracle event ID for sanity
+    ///
+    /// An id within [25h, 23h] from now is considered sane.
+    fn is_oracle_event_id_sane_to_take(&self, now: OffsetDateTime) -> bool {
+        let event_id_timestamp = self.oracle_event_id.timestamp();
+
+        let in_23_hours = now + SETTLEMENT_INTERVAL - Duration::HOUR;
+        let in_25_hours = now + SETTLEMENT_INTERVAL + Duration::HOUR;
+
+        event_id_timestamp >= in_23_hours && event_id_timestamp <= in_25_hours
```

Comment:
👍 I'm okay with this, even though I would probably bite the bullet and go with `settlement_interval_minus_one_hour`. This definitely reads better!

Reply:
I totally get what you mean, but I would still keep it with the specific hours for readability. The main question is: Are we planning to change this constant any time soon? If so we should probably keep this more generic 😬

Reply:
I'm actually tempted to change it ever since we had the issue with the oracle being down for more than >24h. 

Once we have dynamic liquidation we definitely can go longer here.

Reply:
fair enough, I changed it.